packages/form-upload/Upload.js

```diff
@@ -37,6 +58,23 @@ const Upload = ({
 
   const fieldValue = Array.isArray(field.value) ? field.value : [];
 
+  useEffect(() => {
+    if (rest.fileDeliveryOnSubmit && isSubmitting === true) {
```

Comment:
This is a good start :) We're gonna need to rely on more than just `isSubmitting`, I'll find a more fleshed out example with formik, but we want to make sure the form has been validated (which is happening during `isSubmitting`) and doesn't have any errors before we start submitting the uploads